src/transformers/trainer.py

```diff
@@ -3041,7 +3041,7 @@ def evaluation_loop(
         prediction_loss_only = prediction_loss_only if prediction_loss_only is not None else args.prediction_loss_only
 
         # if eval is called w/o train, handle model prep here
-        if self.is_deepspeed_enabled and self.model_wrapped is self.model:
+        if self.is_deepspeed_enabled and not self.deepspeed:
```

Comment:
```suggestion
        if self.is_deepspeed_enabled and self.deepspeed is None:
```
Don't rely on Python magic bool conversion but explicitly check what you want (not sure it's the suggestion above, but you cna fix if necessary :-) )

Reply:
Got it, Thanks!